scripts/dumps/dump_gene_set_from_core.pl

```diff
@@ -72,18 +79,19 @@ =head1 OPTIONS
 use Getopt::Long;
 use Pod::Usage;
 
-my ($dbname, $host, $port, $gene_set_dump_file, $help);
+my ($dbname, $host, $port, $gene_set_dump_file, $id_type, $help);
 
 GetOptions(
     'core-db|core_db=s' => \$dbname,
     'host=s'            => \$host,
     'port=s'            => \$port,
     'outfile=s'         => \$gene_set_dump_file,
+    'id_type=s'         => \$id_type,
     'h|help'            => \$help
 );
 
 pod2usage(1) if $help;
-unless ($dbname and $host and $port and $gene_set_dump_file) {
+unless ($dbname and $host and $port and $gene_set_dump_file and grep( /^$id_type$/, qw(gene protein))) {
```

Comment:
This looks good, but using a variable in a regex like this has the potential to produce unexpected results.

I would suggest either applying the [quotemeta](https://perldoc.perl.org/functions/quotemeta) function with `\Q` and `\E`...
```
unless ($dbname and $host and $port and $gene_set_dump_file and grep( /^\Q$id_type\E$/, qw(gene protein))) {
```

...or turning things around and using the original regex directly as a test condition:
```
unless ($dbname and $host and $port and $gene_set_dump_file and $id_type =~ /^(?:gene|protein)$/) {
```